src/stash/src/postgres.rs

```diff
@@ -372,46 +373,68 @@ impl Postgres {
     where
         I: Iterator<Item = (Id, &'a Antichain<Timestamp>)>,
     {
-        for (collection_id, new_upper) in seals {
-            let upper = Self::upper_tx(stmts, tx, collection_id).await?;
-            if PartialOrder::less_than(new_upper, &upper) {
-                return Err(StashError::from(format!(
-                    "seal request {} is less than the current upper frontier {}",
-                    AntichainFormatter(new_upper),
-                    AntichainFormatter(&upper),
-                )));
-            }
-            tx.execute(&stmts.seal, &[&new_upper.as_option(), &collection_id])
-                .await?;
-        }
+        let futures = seals.map(|(collection_id, new_upper)| {
+            try_join(
+                async move {
+                    let upper = Self::upper_tx(stmts, tx, collection_id).await?;
+                    if PartialOrder::less_than(new_upper, &upper) {
+                        return Err(StashError::from(format!(
+                            "seal request {} is less than the current upper frontier {}",
+                            AntichainFormatter(new_upper),
+                            AntichainFormatter(&upper),
+                        )));
+                    }
+                    Ok(())
+                },
+                async move {
+                    tx.execute(&stmts.seal, &[&new_upper.as_option(), &collection_id])
+                        .map_err(StashError::from)
+                        .await
+                },
+            )
```

Comment:
a quick note so we know whats going on: `try_join` will drop the other futures being joined if one future ends in an error. This is basically equivalent to the old code, but if the old futures had multiple await points, without a `?` on the first await, it could result in slightly different semantics, where a future could be cancelled midway through where it previously wouldn't have

Reply:
Yes, that's ok because this whole thing executes in a postgres transaction, and anything that returns an error will abort it, so all futures can be dropped.